src/xcode/ENA/ENA/Source/Scenes/AntigenTestProfile/AntigenTestProfileViewController.swift

```diff
@@ -62,20 +67,145 @@ class AntigenTestProfileViewController: UIViewController, FooterViewHandling, Di
 		dismiss()
 	}
 
-	// MARK: - Public
+	// MARK: - UITableViewDataSource
+
+	func numberOfSections(in tableView: UITableView) -> Int {
+		viewModel.numberOfSections
+	}
+
+	func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+		viewModel.numberOfItems(in: AntigenTestProfileViewModel.TableViewSections.map(section))
+	}
+
+	// MARK: - UITableViewdelegate
+
+	func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+		guard didCalculateGradientHeight == false,
+			  AntigenTestProfileViewModel.TableViewSections.map(indexPath.section)  == .QRCode else {
+			return
```

Comment:
Perhaps leave a log entry here?

Reply:
I added a comment - a log it's helpful here